### ISSUE TITLE
The config param of the linter.verify() method doesn't support "extends" property

### ISSUE THREAD

**Issue by EvgenyOrekhov** [eslint/eslint#7108]
**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** 6.5.0
* **npm Version:** 3.10.6

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{"extends": "eslint:recommended"}
```

**What did you do? Please include the actual source code causing the issue.**
```js
const linter = require("eslint").linter;
const warnings = linter.verify("console.log();", {"extends": "eslint:recommended"});
```

**What did you expect to happen?**
I expected `linter.verify()` to return an array with two warnings:
- Unexpected console statement (no-console)
- Newline required at end of file but not found (eol-last)

**What actually happened? Please include the actual, raw output from ESLint.**
`linter.verify()` returned `[]` (an empty array)

**Comment by EvgenyOrekhov** [comment]
Also
```js
linter.verify("console.log();", {"extends": "blablablabla"});
```
should throw an error saying `Cannot find module 'eslint-config-blablablabla'`, but instead it just executes successfully and returns an array with the "Undeclared 'console'" warning.

**Comment by gyandeeps** [comment]
`require("eslint").linter` : Doesnt do any config evaluations. Its a very simple and flat api which takes code and config and runs eslint on it.

`require("eslint").CLIEngine` : On the other hand, CLIEngine has all the smarts build into it so that it can evaluate and run thorugh all the config magic like plugins, eslintrc files, etc.

Docs: http://eslint.org/docs/developer-guide/nodejs-api

**Comment by EvgenyOrekhov** [comment]
@gyandeeps these docs say the following about the `config` parameter of the linter:

> config - a configuration object that is equivalent to an eslintrc file

which is not true.

Either the docs should be fixed or the linter should do config evaluations.
I would prefer the second option.

**Comment by platinumazure** [comment]
We can fix the docs. The capabilities of each object in the Node API follow from our architectural design, however, and configuration processing belongs to CLIEngine rather than the linter.

**Comment by EvgenyOrekhov** [comment]
I wish CLIEngine had an `evaluateConfig()` method that would receive a non-evaluated config object.
I think [this function](https://github.com/eslint/eslint/blob/58e6d765494902aba61ae76869f0f010339ddb68/lib/config.js#L43-L69) does what I need.
I would love it to be a part of the public API.

**Comment by platinumazure** [comment]
@EvgenyOrekhov Could you please elaborate on what problem you are trying to solve? Odds are, CLIEngine lets you solve it with the right configuration options.

**Comment by EvgenyOrekhov** [comment]
@platinumazure I have an ESLint config loaded from `.eslintrc.json` and an array of strings of JavaScript sources. I need to map those sources to ESLint warnings.
```js
const linter = require("eslint").linter;
const config = {"extends": "eslint:recommended"};
const sources = [
    "console.log('test');", // source code of a .js file
    "function myFunc() { ..." // source code of another .js file
];
const warnings = sources.map((source) => linter.verify(source, config));
```
The problem is that `linter.verify()` doesn't evaluate the config and `extends` doesn't work here.

**Comment by platinumazure** [comment]
Hi @EvgenyOrekhov, okay, then you want to use CLIEngine's [`executeOnText()`](http://eslint.org/docs/developer-guide/nodejs-api.html#executeontext) method to run the linter against your source text strings.

Here's a rough idea of how you could do this:

```js
const CLIEngine = require("eslint").CLIEngine;
const config = {"extends": "eslint:recommended"};
const sources = [
    "console.log('test');", // source code of a .js file
    "function myFunc() { ..." // source code of another .js file
];

cliEngine = new CLIEngine({
    baseConfig: config,
    useEslintrc: false,   // Assuming that you don't want any .eslintrc in the filesystem to interfere. If you want .eslintrc, omit this
});

const warnings = sources.map((source) => cliEngine.executeOnText(source));
```

That should hopefully be very close to what you want. If you need to tweak the configuration options due to unexpected output or any other reason, take a good look at the many options that can be passed to the [CLIEngine constructor](http://eslint.org/docs/developer-guide/nodejs-api.html#cliengine).

Hope this helps-- if not, feel free to ask another question here or stop by on our [Gitter chat](https://gitter.im/eslint/eslint).

**Comment by EvgenyOrekhov** [comment]
@platinumazure Thank you! Your suggestion solves my problem.
